src/methods.rs

```diff
@@ -1,182 +1,143 @@
 use std::io;
 
+use serde::Deserialize;
 use serde_json::json;
 
 mod chk {
     // this lets us make the RpcMethod trait public but non-implementable by users outside this crate
```

Comment:
BTW, Why do we believe that is necessary? :thinking:

Reply:
It's used repeatedly inside diff method modules for cases where unit types / null is returned from the server. I didn't want to localize it at the time to avoid littering the file with `use` statements. But I'm currently refactoring so we might not have to anymore.

Reply:
Hmm, I feel I don't understand your answer or you did not understand my question :smile: 

I meant, why do we need to make RpcMethod non-implementable by users outside of this crate?

Reply:
Oh yeah, sorry.. I was looking at the `serde::Deserialize` highlight 😅 

Well, for one, we have a finite space possible `RpcMethod` types, so I thought if they were completely exhausted in definition within this crate, we don't need to make them implementable outside.